src/Microsoft.IdentityModel.Logging/TextWriterEventListener.cs

```diff
@@ -90,7 +90,7 @@ protected override void OnEventWritten(EventWrittenEventArgs eventData)
 
             for (int i = 0; i < eventData.Payload.Count; i++)
             {
-                _streamWriter.WriteLine(eventData!.Payload[i]!.ToString());
+                _streamWriter.WriteLine(eventData!.Payload[i]?.ToString());
```

Comment:
Although it might look simpler and nicer it is a bit tricky for two reasons:
1. you have to know that streamewriter.WriteLine() allows arguments to be null
2. writing a null would write a new empty - line - this can affect the logs -> empty lines do add up if the data stream is big it could become something like \r\n\r\n\r\n\r\n in extreme cases being pretty long and big and this being an event listener for logging, could end up being the case
2. it is allways quicker to evaluate a boolean expression than call a function, thus a code that would take more lines of code if (eventData == null) continue;  if (eventData.Payload[i] == null) continue; would still be more performant and would produce less costly logs than nullables over here.

Reply:
feel free to propose a PR w/the changes you are suggesting.